### PR TITLE
Make stream function non-blocking

### PR DESCRIPTION
**Pull request by rushton**

### CODE REVIEW

File: s3wrapper/s3.go [hunk 1]
```diff
@@ -215,6 +216,7 @@ func (w *S3Wrapper) GetAll(keys chan *ListOutput) chan *ListOutput {
 				dir := strings.Join(parts[0:len(parts)-1], "/")
 				util.CreatePathIfNotExists(dir)
 				reader, err := w.GetReader(*k.Bucket, *k.Key)
+				defer reader.Close()
```
Comment (hasjohnb) [inline s3wrapper/s3.go]: I'd put this below the `if err != nil` below -- it's moot if there's an error.

File: s3wrapper/s3.go [hunk 2]
```diff
@@ -157,46 +157,47 @@ func (w *S3Wrapper) GetReader(bucket string, key string) (io.ReadCloser, error)
...
-				panic(err)
-			}
-			ext_reader, err := util.GetReaderByExt(reader, *key.Key)
-			if err != nil {
-				panic(err)
-			}
-
-			for {
-				line, _, err := ext_reader.ReadLine()
+	go func() {
```
Comment (hasjohnb) [inline s3wrapper/s3.go]: Does someone wait on this function (waiting for `lines` to be closed maybe?), or is there a way for the program to exit before it finishes listing?

**Comment by hasjohnb** [comment]
Minor comments, and bump the version. LGTM.

**Comment by rushton** [comment]
tag `v1.2.3` pushed

**Pull request merged**